src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,518 @@
+        }
+    }
+
+    /**
+     * Deletes a specific task
+     * If task number is not provided or invalid, prints an error
+     *
+     * @param params String in the format "delete X", where X is supposed to be the task number
+     */
+    public static void deleteTask(String params) {
+        if (tasks.isEmpty()) {
+            //error if user does not have any tasks to be deleted
+            showMessageFramedWithDivider(MESSAGE_NO_TASKS_YET);
+        } else if (params.equals(EMPTY)) {
+            //error if user inputs only "delete" with no number behind
+            showMessageFramedWithDivider(MESSAGE_ERROR_INVALID_COMMAND_DELETE_FORMAT);
+        } else {
+            try {
+                int taskNum = Integer.parseInt(params);
+                if (taskNum > tasks.size() || taskNum < 1) {
+                    //error if user inputs a task number that does not exist
+                    showMessageFramedWithDivider("Please input a valid task number from 1 to " + tasks.size() + "!");
+                } else {
+                    final String deletedTaskString = tasks.get(taskNum-1).toString();
+                    tasks.remove(taskNum-1);
+                    rewriteTaskListToFile();
+                    showMessageFramedWithDivider("Alrightys! I have removed the following task:", deletedTaskString,
+                            "Current number of tasks: " + tasks.size());
+                }
+            } catch (NumberFormatException e) {
+                //error if user did not input a valid integer for task number
+                showMessageFramedWithDivider(MESSAGE_ERROR_INVALID_COMMAND_DELETE_FORMAT);
+            }
+        }
+    }
+
+    /**
+     * Mark a specific task as done
+     * If task number is not provided or invalid, prints an error
+     *
+     * @param params String in the format "done X", where X is supposed to be the task number
+     */
+    public static void markTaskAsDone(String params) {
+        if (tasks.isEmpty()) {
+            //error if user does not have any tasks to be marked completed
+            showMessageFramedWithDivider(MESSAGE_NO_TASKS_YET);
+        } else if (params.equals(EMPTY)) {
+            //error if user inputs only "done" with no number behind
+            showMessageFramedWithDivider(MESSAGE_ERROR_INVALID_COMMAND_DONE_FORMAT);
+        } else {
+            try {
+                int taskNum = Integer.parseInt(params);
+                if (taskNum > tasks.size() || taskNum < 1) {
+                    //error if user inputs a task number that does not exist
+                    showMessageFramedWithDivider("Please input a valid task number from 1 to " + tasks.size() + "!");
+                } else {
+                    showMessage(DIVIDER);
+                    showMessage(tasks.get(taskNum-1).markAsDone());
+                    rewriteTaskListToFile();
+                    showMessage(tasks.get(taskNum-1).toString(), DIVIDER);
```

Comment:
You might want to [avoid complicated expressions](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-complicated-expressions) here, possibly by using some variables.